### DISCUSSION TITLE
Why subject is null on SoLiD LDP Container?

### DISCUSSION THREAD

**Discussion by linonetwo** [comunica/comunica#446] [Q&A] [answered]
#### Issue type:

- :question: Question

____
#### Question:

I'm trying to read https://linonetwo.solidweb.org/public/aaa/

```turtle
@prefix : <#>.
@prefix n0: <>.
@prefix ldp: <http://www.w3.org/ns/ldp#>.
@prefix terms: <http://purl.org/dc/terms/>.
@prefix XML: <http://www.w3.org/2001/XMLSchema#>.
@prefix st: <http://www.w3.org/ns/posix/stat#>.

n0:
    a ldp:BasicContainer, ldp:Container;
    terms:modified "2019-05-02T15:00:21Z"^^XML:dateTime;
    st:mtime 1556809221.552;
    st:size 4096.
```

Using https://github.com/linonetwo/solid-tiddlywiki-syncadaptor/blob/4a771f8dc9b607ef7df0060ecec28f9341382f4a/src/SyncAdaptor.js#L86-L88
```js
result.bindingsStream.on('data', data => {
      console.log(data.get('?s').value, data.get('?p').value, data.get('?o').value);
});
```

Result is:

```
null http://www.w3.org/1999/02/22-rdf-syntax-ns#type http://www.w3.org/ns/ldp#BasicContainer
null http://www.w3.org/1999/02/22-rdf-syntax-ns#type http://www.w3.org/ns/ldp#Container
null http://purl.org/dc/terms/modified 2019-05-02T15:00:20Z
null http://www.w3.org/ns/posix/stat#mtime 1556809220.54
null http://www.w3.org/ns/posix/stat#size 4096
```

Expected behavior:

Actually I don't quite know, maybe it could be `/` instead of `null`? Or a base url provided by me? (If I can provide a base URI as the Graph part or quad?)

**Comment by linonetwo** [comment]
Is this related to https://github.com/comunica/comunica/issues/311 ?

**Comment by rubensworks** [comment, marked as answer]
Terms should never be able to become `null`. Can you check what happens when you just print the parsed quads?

It might be possible that some terms in there are `null` already, and that you should indeed pass a `baseIRI` to the parser.

**Comment by linonetwo** [comment]
Sure, `subject: Object { id: "null" }` from N3Parser.

Provide idp as baseIRI solved this:

```js
const session: SoLiDSession = await solidAuthClient.currentSession();
const parser = new Parser({ format: 'Turtle', baseIRI: session.idp });
```
